### ISSUE TITLE
mpi job not support hostNetwork: true pattern

### ISSUE THREAD

**Issue by tongchao199** [volcano-sh/volcano#778]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>

/kind bug

**What happened**:
When I ran [mpi-example](https://github.com/volcano-sh/volcano/blob/release-0.4/example/integrations/mpi/mpi-example.yaml), I used the `hostNetwork:true` pattern and found that the task could not parse the mpiworker.host
```shell
root@gpu-ser348-06:/etc/volcano# cat mpiworker.host  && echo
lm-mpi-job-mpiworker-0.lm-mpi-job
lm-mpi-job-mpiworker-1.lm-mpi-job
lm-mpi-job-mpiworker-2.lm-mpi-job
root@gpu-ser348-06:/etc/volcano# ping lm-mpi-job-mpiworker-0.lm-mpi-job
ping: unknown host lm-mpi-job-mpiworker-0.lm-mpi-job
```

**What you expected to happen**:
Use the `hostNetwork:true` pattern could  parse the mpiworker.host

**How to reproduce it (as minimally and precisely as possible)**:
* Deploy volcano
* Modify [mpi-example](https://github.com/volcano-sh/volcano/blob/release-0.4/example/integrations/mpi/mpi-example.yaml) and apply it
```yaml
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  name: lm-mpi-job
spec:
  minAvailable: 3
  schedulerName: volcano
  plugins:
    ssh: []
    svc: []
  tasks:
    - replicas: 1
      name: mpimaster
      policies:
        - event: TaskCompleted
          action: CompleteJob
      template:
        spec:
          hostNetwork: true
          containers:
            - command:
                - /bin/sh
                - -c
                - |
                  MPI_HOST=`cat /etc/volcano/mpiworker.host | tr "\n" ","`;
                  mkdir -p /var/run/sshd; /usr/sbin/sshd;
                  mpiexec --allow-run-as-root --host ${MPI_HOST} -np 2 mpi_hello_world > /home/re;
              image: volcanosh/example-mpi:0.0.1
              name: mpimaster
              ports:
                - containerPort: 22
                  name: mpijob-port
              workingDir: /home
          restartPolicy: OnFailure
    - replicas: 2
      name: mpiworker
      template:
        spec:
          hostNetwork: true
          containers:
            - command:
                - /bin/sh
                - -c
                - |
                  mkdir -p /var/run/sshd; /usr/sbin/sshd -D;
              image: volcanosh/example-mpi:0.0.1
              name: mpiworker
              ports:
                - containerPort: 22
                  name: mpijob-port
              workingDir: /home
          restartPolicy: OnFailure
---
```

**Anything else we need to know?**:

**Environment**:
- Volcano Version:
```
kubectl -n volcano-system get pods -o json | jq '.items[].spec.containers[0] | "name:" + .name + " image:" + .image'
"name:admission image:volcanosh/vc-webhook-manager:v0.4"
"name:main image:volcanosh/vc-webhook-manager:v0.4"
"name:volcano-controllers image:volcanosh/vc-controller-manager:v0.4"
"name:volcano-scheduler image:volcanosh/vc-scheduler:v0.4"
```
- Kubernetes version (use `kubectl version`):
```shell
kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14-mlpe-20200217", GitCommit:"883cfa7a769459affa307774b12c9b3e99f4130b", GitTreeState:"clean", BuildDate:"2020-02-17T14:06:28Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
```shell
    description: Computer
    product: KVM
    vendor: Red Hat
    version: RHEL 7.0.0 PC (i440FX + PIIX, 1996)
```
- OS (e.g. from /etc/os-release):
```shell
root@gpu-ser348-06:~$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):
```shell
root@gpu-ser348-06:~$ uname -a
Linux gpu-ser348-06 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Others:

**Comment by k82cn** [comment]
/cc @hzxuzhonghu 

It's better to handle this feature by job controller :)

**Comment by hzxuzhonghu** [comment]
@tongchao199 You need to set the dnspolicy to `ClusterFirstWithHostNet` when you are using hostnetwork

**Comment by hzxuzhonghu** [comment]
We should set this for user when dnsPolicy is not specified in this case.

**Comment by tongchao199** [comment]
> @tongchao199 You need to set the dnspolicy to `ClusterFirstWithHostNet` when you are using hostnetwork

thanks, The domain name of the mpiworker.host file can be resolved normally

**Comment by k82cn** [comment]
let's also put this into FAQ :)

**Comment by hzxuzhonghu** [comment]
I meant to set `ClusterFirstWithHostNet ` by default when explicitly set hostnetwork

**Comment by k82cn** [comment]
> I meant to set `ClusterFirstWithHostNet ` by default when explicitly set hostnetwork

ok :)

**Comment by k82cn** [comment]
what's next step? Should be fixed by #779?

**Comment by tongchao199** [comment]
Fixed with #779
